subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomResolveIntegrationTest.groovy

```diff
@@ -52,7 +52,6 @@ task retrieve(type: Sync) {
 
         and:
         original.pom.expectGet()
-        original.artifact.expectHead()
```

Comment:
This test case is to test `relocation` element of `pom.xml`, where`groupA:projectA:1.2`(original) is relocated to `newGroupA:projectA:1.2`(newModule). 

Before this PR, the following events will happen:

- `GET` `groupA:projectA:1.2`'s pom 
- `HEAD` `groupA:projectA:1.2`'s artifact (jar)
- `GET` `newGroupA:projectA:1.2`'s pom
- `GET` `newGroupA:projectA:1.2`'s artifact (jar)

The issue is that `groupA:projectA:1.2`'s artifact should not be retrieved since it has been relocated. However, since `groupA:projectA:1.2` and `newGroupA:projectA:1.2`'s artifact names are same (both `projectA.jar`), this test didn't cover the #1789 scenario.

What this PR does is that it will prevent the retrieving of original artifacts if `<relocation>` element exists. So, after this PR, the events will be:

- `GET` `groupA:projectA:1.2`'s pom 
- `GET` `newGroupA:projectA:1.2`'s pom
- `GET` `newGroupA:projectA:1.2`'s artifact

That's why the `HEAD` was removed.

Moreover, inspired by your [previous advice](https://github.com/gradle/gradle/pull/2166/commits/190e808314a8e6d771249c6b5021906768803c30) that we shouldn't depend on an external resource, I found the original extra integration test can be merged into this one, so I modified this one slightly to cover #1789 scenario.